validator/accounts/accounts_delete.go

```diff
@@ -102,37 +101,28 @@ func DeleteAccountCli(cliCtx *cli.Context) error {
 
 // DeleteAccount deletes the accounts that the user requests to be deleted from the wallet.
 func DeleteAccount(ctx context.Context, cfg *Config) error {
-	switch cfg.Wallet.KeymanagerKind() {
-	case keymanager.Remote:
-		return errors.New("cannot delete accounts for a remote keymanager")
-	case keymanager.Imported:
-		km, ok := cfg.Keymanager.(*imported.Keymanager)
-		if !ok {
-			return errors.New("not a imported keymanager")
-		}
-		if len(cfg.DeletePublicKeys) == 1 {
-			log.Info("Deleting account...")
-		} else {
-			log.Info("Deleting accounts...")
-		}
-		if err := km.DeleteAccounts(ctx, cfg.DeletePublicKeys); err != nil {
-			return errors.Wrap(err, "could not delete accounts")
-		}
-	case keymanager.Derived:
-		km, ok := cfg.Keymanager.(*derived.Keymanager)
-		if !ok {
-			return errors.New("not a derived keymanager")
-		}
-		if len(cfg.DeletePublicKeys) == 1 {
-			log.Info("Deleting account...")
-		} else {
-			log.Info("Deleting accounts...")
-		}
-		if err := km.DeleteAccounts(ctx, cfg.DeletePublicKeys); err != nil {
-			return errors.Wrap(err, "could not delete accounts")
+	deleter, ok := cfg.Keymanager.(keymanager.Deleter)
+	if !ok {
+		return errors.New("keymanager does not implement Deleter interface")
+	}
+	if len(cfg.DeletePublicKeys) == 1 {
+		log.Info("Deleting account...")
+	} else {
+		log.Info("Deleting accounts...")
+	}
+	statuses, err := deleter.DeleteKeystores(ctx, cfg.DeletePublicKeys)
+	if err != nil {
+		return errors.Wrap(err, "could not delete accounts")
+	}
+	for i, status := range statuses {
+		switch status.Status {
+		case ethpbservice.DeletedKeystoreStatus_ERROR:
+			log.Errorf("Error deleting key %#x: %s", bytesutil.Trunc(cfg.DeletePublicKeys[i]), status.Message)
+		case ethpbservice.DeletedKeystoreStatus_NOT_ACTIVE:
+			log.Warnf("Key %#x is not active, so nothing to delete", bytesutil.Trunc(cfg.DeletePublicKeys[i]))
```

Comment:
Discussed offline, this warning can be misleading. The NOT_ACTIVE enum value is only used for duplicate keys in requests.